### ISSUE TITLE
RUF100 incorrectly reports C901 noqa as unused when C90 category is enabled

### ISSUE THREAD

**Issue by adrian-begg-ba** [astral-sh/ruff#24004]
## Summary

RUF100 (unused noqa directive) incorrectly reports `# noqa: C901` as "unused" when the mccabe complexity rule is enabled via the category `C90` rather than the specific code `C901`.

## Reproduction

**pyproject.toml:**
```toml
[tool.ruff.lint]
select = [
    "C90",    # mccabe complexity
]

[tool.ruff.lint.mccabe]
max-complexity = 10
```

**test.py:**
```python
def complex_function(a, b, c, d, e):  # noqa: C901
    if a:
        if b:
            if c:
                if d:
                    if e:
                        return 1
                    return 2
                return 3
            return 4
        return 5
    return 6
```

**Commands and output:**

```bash
# C901 is being enforced (noqa suppresses it)
$ ruff check --select C901 test.py
All checks passed!

# Without noqa, C901 fires
$ ruff check --select C901 --ignore-noqa test.py
C901 \`complex_function\` is too complex (6 > 10)

# But RUF100 says the noqa is unused
$ ruff check --select RUF100 test.py
RUF100 [*] Unused \`noqa\` directive (non-enabled: \`C901\`)
```

## Expected behavior

RUF100 should recognize that `C901` is enabled when `C90` (the mccabe category) is in the select list, and NOT report the noqa as unused.

## Actual behavior

RUF100 reports `(non-enabled: \`C901\`)` even though the rule IS enabled via the `C90` category and the noqa IS actively suppressing a violation.

## Impact

Running `ruff check --fix` or `ruff check --select RUF100 --fix` incorrectly removes valid C901 noqa comments, causing new C901 violations to appear.

## Environment

```
$ ruff --version
ruff 0.11.4
```

## Workaround

Change `select = ["C90"]` to `select = ["C901"]` - but this is less flexible and requires knowing all specific rule codes rather than using categories.

**Comment by amyreese** [comment]
This is because `--select RUF100` overrides your configured `select = ["C90"]` in your `pyproject.toml`, and so the *only* rule enabled is `RUF100`. At that point, `RUF100` sees that no diagnostic was generated, and treats the `#noqa` as unused.

If you instead run with `--select C901,RUF100` you will see the output I assume you expect:

```
$ cargo -q run -p ruff -- check --no-cache --select RUF100,C901 ../test.py
All checks passed!
```

I think what you *really* want though is `--extend-select RUF100`, which enables `RUF100` *in addition to* the rules you normally have configured in your project config:

```
$ cargo -q run -p ruff -- check --no-cache --select C901 --ignore-noqa ../test2.py
C901 `complex_function` is too complex (6 > 5)
 --> /Users/amethyst/workspace/test2.py:1:5
  |
1 | def complex_function(a, b, c, d, e):  # noqa: C901
  |     ^^^^^^^^^^^^^^^^
2 |     if a:
3 |         if b:
  |

Found 1 error.
> [1]

$ cargo -q run -p ruff -- check --no-cache --extend-select RUF100 ../test2.py
All checks passed!
```

**Comment by adrian-begg-ba** [comment]
Thank you for the clarification! I was using `--select RUF100` which overrode my config instead of `--extend-select RUF100`.

Using `ruff check --extend-select RUF100 --fix` correctly identified unused noqa directives.
